src/compiler/checker.ts

```diff
@@ -7261,7 +7261,7 @@ namespace ts {
                 if (links.jsxFlags & JsxFlags.ClassElement) {
                     let elemInstanceType = getJsxElementInstanceType(node);

```

Comment:
@RyanCavanaugh 	just tried the solution we discussed in #3960 and it fails to:

```ts
import * as React from 'react';

class C extends React.Component<any, any> {
    render() {
        return null
    }
}

export type ReactCtor<P, S> = new() => React.Component<P, S> & { render(): React.ReactElement<P> };
let C1: ReactCtor<any, any> = C;
let a = <C1 />;
```

So maybe the error is not because of call or construct signature?